Lib/multiprocessing/queues.py

```diff
@@ -45,21 +46,37 @@ def __init__(self, maxsize=0, *, ctx):
         else:
             self._wlock = ctx.Lock()
         self._sem = ctx.BoundedSemaphore(maxsize)
+
+        self._lock_shutdown = ctx.Lock()
+        # Cannot use a ctx.Value because 'ctypes' library is
+        # not always available on all Linux platforms.
+        # Use of Semaphores instead of an array from `heap.BufferWrapper'
+        # is here more efficient and explicit.
+        self._sem_flag_shutdown = ctx.Semaphore(0)
+        self._sem_flag_shutdown_immediate = ctx.Semaphore(0)
```

Comment:
We found in the threading and async queues that it was simpler to implement and reason about when we only have a single boolean flag for queue-shutdown state: all `immediate` did was consume all of the items in the queue (and subtracted the count from unfinished task count). Is that possible here?